feat: extract core booking audit infrastructure from PR 25125

hariombalhara opened the pull request:
## What does this PR do?

Extracts the core booking audit infrastructure from PR #25125 into a separate branch, **excluding integration changes with booking flows**. This allows the foundational audit system to be reviewed and merged independently.

**Included:**
- All `packages/features/booking-audit/*` (action services, repository, task consumer, viewer service)
- `packages/features/di/containers/BookingAuditViewerService.container.ts` (DI container)
- `packages/features/tasker/tasker.ts` (audit task type updates)
- `packages/features/bookings/lib/types/actor.ts` (actor types + `makeAttendeeActor`)
- `packages/features/bookings/repositories/BookingRepository.ts` (`getFromRescheduleUid` method)
- `apps/web/modules/booking/logs/views/booking-logs-view.tsx` (UI for viewing audit logs)
- Translation strings in `common.json`

**Excluded (integration changes):**
- `packages/trpc/server/*` (tRPC handlers)
- `packages/features/ee/round-robin/*` (round-robin integration)
- `packages/features/bookings/lib/handleCancelBooking.ts`
- `packages/features/bookings/lib/handleConfirmation.ts`
- `packages/features/bookings/lib/onBookingEvents/BookingEventHandlerService.ts`
- `packages/features/bookings/lib/service/RegularBookingService.ts`
- `apps/api/v2/*` (API v2 integration)

## Mandatory Tasks (DO NOT REMOVE)

- [x] I have self-reviewed the code (A decent size PR without self-review might be rejected).
- [x] I have updated the developer docs in /docs if this PR makes changes that would require a [documentation change](https://cal.com/docs). N/A - infrastructure only
- [ ] I confirm automated tests are in place that prove my fix is effective or that my feature works.

## How should this be tested?

This PR contains **infrastructure only** - the audit system won't be triggered until integration changes from PR #25125 are merged separately. To verify:

1. Ensure the code compiles: `yarn type-check:ci --force`
2. Verify no integration files were accidentally included
3. The booking logs UI at `/booking/logs/[uid]` should render (though no logs will appear without integration)

## Checklist for Human Review

- [ ] Verify excluded files are correct (no integration changes leaked in)
- [ ] Review type change from discriminated union to lean base schema (`BookingAuditTaskBaseSchema` with `data: z.unknown()`)
- [ ] Check `BookingAuditViewerService.checkPermissions()` - throws error in production (TODO comment)
- [ ] Review eslint-disable comment added for intentional switch fall-through in `BookingAuditTaskConsumer.ts`

---

Link to Devin run: https://app.devin.ai/sessions/349411f11d51483494218c992d2b9220
Requested by: hariom@cal.com (@hariombalhara)

devin-ai-integration[bot] committed:
feat: extract core booking audit infrastructure from PR 25125

This PR contains only the core booking audit infrastructure changes from PR 25125,
excluding integration changes with booking flows.

Included:
- All packages/features/booking-audit/* (core audit services, actions, repository)
- packages/features/di/containers/BookingAuditViewerService.container.ts
- packages/features/tasker/tasker.ts (audit task types)
- packages/features/bookings/lib/types/actor.ts (actor types for audit)
- packages/features/bookings/repositories/BookingRepository.ts (getFromRescheduleUid method)
- apps/web/modules/booking/logs/views/booking-logs-view.tsx (UI for viewing audit logs)
- apps/web/public/static/locales/en/common.json (translations)

Excluded (integration changes):
- packages/trpc/server/* (tRPC handlers)
- packages/features/ee/round-robin/* (round-robin integration)
- packages/features/bookings/lib/handleCancelBooking.ts
- packages/features/bookings/lib/handleConfirmation.ts
- packages/features/bookings/lib/onBookingEvents/BookingEventHandlerService.ts
- packages/features/bookings/lib/service/RegularBookingService.ts
- apps/api/v2/* (API v2 integration)

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

hariombalhara commented on apps/web/modules/booking/logs/views/booking-logs-view.tsx:
Booking History UI improvement.
- JSON styling
- Reschedule Title with dynamic values and link support
- Other UI changes

devin-ai-integration[bot] committed:
fix: make booking audit interfaces backwards-compatible with main

- Add queueAudit method back to BookingAuditProducerService interface for backwards compatibility
- Implement queueAudit method in BookingAuditTaskerProducerService
- Make userTimeZone parameter optional in BookingAuditViewerService
- Add BookingAuditTaskProducerActionData type for legacy queueAudit method
- Use any generics in BookingAuditActionServiceRegistry (matching PR 25125)
- Fix type assertions in BookingAuditTaskConsumer

Co-Authored-By: hariom@cal.com <hariombalhara@gmail.com>

hariombalhara committed:
fix switch eslint and ts

Udit-takkar commented on apps/web/modules/booking/logs/views/booking-logs-view.tsx:
```diff
@@ -70,14 +85,26 @@ const getActionIcon = (action: string) => {
+        default:
+            return null;
+    }
+};
```
We should use enum type for actorType

Udit-takkar commented on apps/web/public/static/locales/en/common.json:
```diff
@@ -4095,8 +4095,25 @@
+    "rejected": "Rejected",
+    "attendee_removed": "Attendee Removed",
+    "reassignment": "Reassignment",
+    "booking_reassigned_to_host": "Booking reassigned to {{host}}",
```
booking_reassigned_to_host already exists with value "Booking reassigned to {{host}}"

Udit-takkar commented on packages/features/booking-audit/lib/service/BookingAuditViewerService.ts:
```diff
@@ -95,26 +103,107 @@ export class BookingAuditViewerService {
+
+        if (!rescheduledLog) {
+            // TODO: Use logger instead of console.error
+            console.error(`No rescheduled log found for booking ${fromRescheduleUid} -> ${currentBookingUid}`);
```
can we use logger here?

hariombalhara committed:
feat: enhance BookingAuditViewerService with logging and type improvements

- Added ISimpleLogger dependency to BookingAuditViewerService for better error handling.
- Updated actor type in enriched audit logs to use AuditActorType for improved type safety.
- Replaced console.error with logger for error reporting when no rescheduled log is found.

Udit-takkar committed:
Merge branch 'main' into devin/booking-audit-core-1765285076

hariombalhara commented on apps/web/modules/booking/logs/views/booking-logs-view.tsx:
```diff
@@ -70,14 +86,26 @@ const getActionIcon = (action: string) => {
+        default:
+            return null;
+    }
+};
```
Oh you are right, it certainly doesn't need switch. I will fix it in a followup PR
